### PR TITLE
[SPARK-10490] [ML] Consolidate the Cholesky solvers in WeightedLeastSquares and ALS

### PR DESCRIPTION
**Pull request by yanboliang**
Consolidate the Cholesky solvers in WeightedLeastSquares and ALS.

### CODE REVIEW

**Comment by yanboliang** [comment]
Jenkins, test this please.

File: mllib/src/main/scala/org/apache/spark/mllib/linalg/CholeskyDecomposition.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import com.github.fommil.netlib.LAPACK.{getInstance => lapack}
+import org.apache.spark.annotation.Experimental
+import org.netlib.util.intW
+
+/**
+ * :: Experimental ::
+ * Compute Cholesky decomposition.
+ */
+@Experimental
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/linalg/CholeskyDecomposition.scala]: We don't need `@Experimental` tag for package private classes.

File: mllib/src/main/scala/org/apache/spark/mllib/linalg/CholeskyDecomposition.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.mllib.linalg
+
+import com.github.fommil.netlib.LAPACK.{getInstance => lapack}
+import org.apache.spark.annotation.Experimental
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/linalg/CholeskyDecomposition.scala]: organize imports

File: mllib/src/main/scala/org/apache/spark/mllib/linalg/CholeskyDecomposition.scala [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+private[spark] object CholeskyDecomposition {
+
+  /**
+   * Solves a symmetric positive definite linear system via Cholesky factorization.
+   * The input arguments are modified in-place to store the factorization and the solution.
+   * @param A the upper triangular part of A
+   * @param bx right-hand side
+   * @return the solution array
+   */
+  private[spark] def solve(A: Array[Double], bx: Array[Double]): Array[Double] = {
```
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/linalg/CholeskyDecomposition.scala]: this should be public. The object is already package private.

File: mllib/src/main/scala/org/apache/spark/mllib/linalg/EigenValueDecomposition.scala [hunk 4]
```diff
@@ -21,13 +21,9 @@ import breeze.linalg.{DenseMatrix => BDM, DenseVector => BDV}
 import com.github.fommil.netlib.ARPACK
 import org.netlib.util.{intW, doubleW}
 
-import org.apache.spark.annotation.Experimental
-
 /**
- * :: Experimental ::
  * Compute eigen-decomposition.
  */
-@Experimental
```
Comment (yanboliang) [inline mllib/src/main/scala/org/apache/spark/mllib/linalg/EigenValueDecomposition.scala]: Remove the ```@Experimental``` tag BTW.

**Comment by mengxr** [comment]
LGTM. Merged into master. Thanks!

**Pull request closed without merging**